index.js

```diff
@@ -147,12 +156,24 @@ JSONdb.prototype.deleteAll = function() {
  */
 JSONdb.prototype.sync = function() {
   if (this.options && this.options.asyncWrite) {
-    fs.writeFile(this.filePath, this.options.stringify(this.storage, null, this.options.jsonSpaces), (err) => {
-      if (err) throw err;
+    return new Promise((res, rej) => {
+      if (this.options.stringify === defaultOptions.stringify) {
```

Comment:
Comparing functions this way feels hacky, just a simple `.bind({})` is enough to make this check fail.